R/checks.R

```diff
@@ -24,20 +25,63 @@
 check_user_rights <- function(db_data,
                               db_metadata,
                               call = caller_env()) {
-  missing_db_metadata <- db_metadata %>% # nolint: object_usage_linter
-    filter(!.data$field_name_updated %in% names(db_data)) %>%
+  # Set condition components used in both error and warning
+  msg_info <- c("i" = "This can happen when the user privileges are not set to allow exporting certain
+    instruments via the API.")
+  cond_class <- c("redcap_user_rights", "REDCapTidieR_cond")
+
+  db_fields <- names(db_data)
+  # Fields in metadata excluding record ID
+  metadata_fields <- setdiff(db_metadata$field_name_updated, get_record_id_field(db_data))
+
+  # If no data fields were returned the user has no data access
+  if (length(intersect(db_fields, metadata_fields)) == 0) {
+    cli_abort(
+      c(
+        "x" = "The REDCap API returned no data for any of the requested instruments.",
```

Comment:
I think we should replace all messages that say "instruments" with "forms" since the argument is `forms =` and it's also the column name in the output super tibble (`redcap_form_name`).

Reply:
I'm fine with that. My only hesitation is that we seem to prefer "instrument" in the glossary: https://chop-cgtinformatics.github.io/REDCapTidieR/articles/glossary.html#instrument

Reply:
Yea... I also saw that. My opinion is still that if `forms` is used as the argument, and the error points users to their argument, then it should reflect that. Thoughts from @skadauke?